apps/arweave/src/ar_merkle.erl

```diff
@@ -57,186 +57,93 @@ validate_path(ID, Dest, RightBound, Path, Ruleset) when Dest < 0 ->
-		{strict_data_split_ruleset, PathSize, DataSize}) ->
-	case EndOffset - LeftBound > ?DATA_CHUNK_SIZE
-			orelse RightBound - LeftBound > ?DATA_CHUNK_SIZE of
+		PathSize, _DataSize, IsRightMostInItsSubTree, LeftBoundShift,
+		CheckBorders, CheckSplit, _AllowRebase) ->
+	AreBordersValid = case CheckBorders of
 		true ->
-			false;
+			%% Borders are only valid if every offset does not exceed the previous offset
+			%% by more than ?DATA_CHUNK_SIZE
+			EndOffset - LeftBound =< ?DATA_CHUNK_SIZE andalso
+				RightBound - LeftBound =< ?DATA_CHUNK_SIZE;
 		false ->
+			%% Borders are always valid if we don't need to check them
+			true
+	end,
+	IsSplitValid = case CheckSplit of
+		true ->
+			%% Reject chunks smaller than 256 KiB unless they are the last or the only chunks
+			%% of their datasets or the second last chunks which do not exceed 256 KiB when
+			%% combined with the following (last) chunks. Finally, reject chunks smaller than
+			%% their Merkle proofs unless they are the last chunks of their datasets.
 			ChunkSize = EndOffset - LeftBound,
-			IsSplitValid =
-				case validate_strict_split of
-					_ when ChunkSize == (?DATA_CHUNK_SIZE) ->
-						LeftBound rem (?DATA_CHUNK_SIZE) == 0;
-					_ when EndOffset == DataSize ->
-						Border = RightBound - RightBound rem (?DATA_CHUNK_SIZE),
-						RightBound rem (?DATA_CHUNK_SIZE) > 0
-								andalso LeftBound =< Border;
-					_ when PathSize > ChunkSize ->
-						false;
-					_ ->
-						LeftBound rem (?DATA_CHUNK_SIZE) == 0
-								andalso DataSize - LeftBound > (?DATA_CHUNK_SIZE)
-								andalso DataSize - LeftBound < 2 * (?DATA_CHUNK_SIZE)
-				end,
-			case IsSplitValid of
-				false ->
-					false;
+			case IsRightMostInItsSubTree of
 				true ->
-					case hash([hash(Data), hash(note_to_binary(EndOffset))]) of
-						ID ->
-							{Data, LeftBound, max(min(RightBound, EndOffset), LeftBound + 1)};
-						_ ->
-							false
-					end
-			end
-	end;
-validate_path(ID, Dest, LeftBound, RightBound,
-		<< L:?HASH_SIZE/binary, R:?HASH_SIZE/binary, Note:(?NOTE_SIZE*8), Rest/binary >>,
-		{strict_data_split_ruleset, PathSize, DataSize}) ->
-	case hash([hash(L), hash(R), hash(note_to_binary(Note))]) of
-		ID ->
-			{Path, NextLeftBound, NextRightBound} =
-				case Dest < Note of
-					true ->
-						{L, LeftBound, min(RightBound, Note)};
-					false ->
-						{R, max(LeftBound, Note), RightBound}
-				end,
-			validate_path(Path, Dest, NextLeftBound, NextRightBound, Rest,
-					{strict_data_split_ruleset, PathSize, DataSize});
-		_ ->
+					%% The last chunk may either start at the bucket start or
+					%% span two buckets.
+					Bucket0 = LeftBound div (?DATA_CHUNK_SIZE),
+					Bucket1 = EndOffset div (?DATA_CHUNK_SIZE),
+					(LeftBound rem (?DATA_CHUNK_SIZE) == 0) orelse Bucket0 + 1 == Bucket1;
+				_ ->
+					%% May also be the only chunk of a single-chunk subtree.
+					LeftBound rem (?DATA_CHUNK_SIZE) == 0 andalso PathSize =< ChunkSize
```

Comment:
I added in the PathSize check from `validate_path_strict_data_split`. The rest of the IsSplitValid come from your new `offset_rebase_support_ruleset` code

Reply:
Removed the check!